CMakeLists.txt

```diff
@@ -20,163 +20,46 @@ endif()
-  message ("Fortran compiler: " ${Fortran_COMPILER_NAME})
-  message ("No optimized Fortran compiler flags are known, we just try -O2...")
-  set (CMAKE_Fortran_FLAGS_RELEASE "-O2")
-  set (CMAKE_Fortran_FLAGS_DEBUG   "-O0 -g")
-endif (Fortran_COMPILER_NAME MATCHES "GNU")
-
-if (CMAKE_BUILD_TYPE MATCHES "DEBUG")
-  add_definitions("-DDEBG")
-endif (CMAKE_BUILD_TYPE MATCHES "DEBUG")
-
-
-find_package(MPI REQUIRED)
-# Stop if there is no MPI_Fortran_Compiler
-if (MPI_Fortran_COMPILER)
-    message(STATUS "MPI_Fortran_COMPILER found: ${MPI_Fortran_COMPILER}")
-else (MPI_Fortran_COMPILER)
-    message(SEND_ERROR "This application cannot compile without MPI")
-endif(MPI_Fortran_COMPILER)
-# Warning if Include are not found => can be fixed with more recent cmake version
-if (MPI_FOUND)
-    message(STATUS "MPI FOUND: ${MPI_FOUND}")
-    include_directories(SYSTEM ${MPI_INCLUDE_PATH})
-    message(STATUS "MPI INCL ALSO FOUND: ${MPI_INCLUDE_PATH}")
-else (MPI_FOUND)
-    message(STATUS "NO MPI include have been found. The executable won't be targeted with MPI include")
-    message(STATUS "Code will compile but performaces can be compromised")
-    message(STATUS "Using a CMake vers > 3.10 should solve the problem")
-    message(STATUS "Alternatively use ccmake to manually set the include if available")
-endif (MPI_FOUND)
-
-execute_process(
-  COMMAND git describe --tag --long --always
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GIT_VERSION
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-add_definitions("-DVERSION=\"${GIT_VERSION}\"")
-option(DOUBLE_PRECISION "Build Xcompact with double precision" ON)
-if (DOUBLE_PRECISION)
-  add_definitions("-DDOUBLE_PREC")
-endif()
-
-option(SINGLE_PRECISION_OUTPUT "Build XCompact with output in single precision" OFF)
-if (SINGLE_PRECISION_OUTPUT)
-  add_definitions("-DSAVE_SINGLE")
-endif()
-
-# FFT options
-set(FFT_Choice "generic" CACHE STRING "FFT for XCompact3d project (generic is the default)")
-set_property(CACHE FFT_Choice PROPERTY STRINGS generic fftw3 mkl)
-
-# Look for ADIOS2 library
-option(USE_ADIOS2 "Build XCompact with ADIOS2 library" OFF)
-if(USE_ADIOS2)
-  #find_package(ADIOS2 COMPONENTS fortran)
-  find_package(ADIOS2)
-endif()
-if(ADIOS2_FOUND)
-  message(STATUS "ADIOS INCLUDE ${ADIOS2_INCLUDE_DIRS}")
-  include_directories(${ADIOS2_INCLUDE_DIRS})
-  add_definitions("-DADIOS2")
-else(ADIOS_FOUND)
-  message(STATUS "ADIOS2 not found")
-endif()
-
-# Create a static library for the fft
-add_subdirectory(decomp2d)
-
-# Create the Xcompact3d executable
+string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE)
+if (CMAKE_BUILD_TYPE MATCHES "NONE")
+  message (STATUS "Selected build type : None")
+elseif (CMAKE_BUILD_TYPE MATCHES "RELEASE")
+  message (STATUS "Selected build type : Release")
+elseif (CMAKE_BUILD_TYPE MATCHES "DEBUG")
+  message (STATUS "Selected build type : Debug")
+elseif (CMAKE_BUILD_TYPE MATCHES "DEV")
+  message (STATUS "Selected build type : Dev")
```

Comment:
@pbartholomew08 add a `FATAL_ERROR` message for this build configuration

Reply:
Dev is currently blocked here : https://github.com/xcompact3d/Incompact3d/blob/2decomp_external/cmake/X3D_Compilers.cmake#L52

Reply:
OK, it's double blocked